include/sharg/parser.hpp

```diff
@@ -524,60 +504,56 @@ class parser
 
     /*!\brief Adds an help page section to the sharg::parser.
      * \param[in] title The title of the section.
-     * \param[in] spec Whether to always display this section title (sharg::option_spec::standard), only when showing
-     *                 the advanced help page (sharg::option_spec::advanced) or never (sharg::option_spec::hidden).
+     * \param[in] advanced_only If set to true, the section only shows when the user requested the advanced help page.
      * \details This only affects the help page and other output formats.
      */
-    void add_section(std::string const & title, option_spec const spec = option_spec::standard)
+    void add_section(std::string const & title, bool const advanced_only = false)
     {
         std::visit(
             [&](auto & f)
             {
-                f.add_section(title, spec);
+                f.add_section(title, advanced_only);
             },
             format);
     }
 
     /*!\brief Adds an help page subsection to the sharg::parser.
      * \param[in] title The title of the subsection.
-     * \param[in] spec Whether to always display this subsection title (sharg::option_spec::standard), only when showing
-     *                 the advanced help page (sharg::option_spec::advanced) or never (sharg::option_spec::hidden).
+     * \param[in] advanced_only If set to true, the section only shows when the user requested the advanced help page.
      * \details This only affects the help page and other output formats.
      */
-    void add_subsection(std::string const & title, option_spec const spec = option_spec::standard)
+    void add_subsection(std::string const & title, bool const advanced_only = false)
     {
         std::visit(
             [&](auto & f)
```

Comment:
```suggestion
            [&title, advanced_only](auto & f)
```

Mainly because of consistency :)